lib/utils.js

```diff
@@ -226,11 +226,10 @@ utils.normalizeRequest = function normalizeRequest(request) {
     // For streaming v4 auth, the total body content length
     // without the chunk metadata is sent as
     // the x-amz-decoded-content-length
-    const contentLength = request.headers['x-amz-decoded-content-length'] ?
-        request.headers['x-amz-decoded-content-length'] :
+    request.contentLength = request.headers['x-amz-decoded-content-length'] ||
```

Comment:
I wouldn't add a request.contentLength to the request object.  I would leave this file alone and when you are pulling the body size, I would use request.headers['content-length']